lib/internal/process.js

```diff
@@ -283,6 +283,53 @@ function setupUncaughtExceptionCapture(exceptionHandlerState) {
   };
 }
 
+function setupCompareVersion() {
+  const {
+    majorVersion,
+    minorVersion,
+    patchVersion,
+  } = process.release;
+  const prereleaseTag = process.release.prereleaseTag.replace(/^-/, '');
+
+  process.release.compareVersion =
+    (major, minor, patch, prerelease = false) => {
+      if (typeof major === 'string')
+        [major, minor, patch] = major.split('.');
+      if (/-/.test(patch))
+        [patch, prerelease] = patch.split('-');
```

Comment:
Both `split()` calls are problematic if you are trying to comply with semver, see [section 9 of the spec](https://semver.org/#spec-item-9). The prerelease label can contain dots and dashes.